bundles/core/org.eclipse.smarthome.core.thing.xml/src/main/java/org/eclipse/smarthome/core/thing/xml/internal/ThingTypeXmlResult.java

```diff
@@ -75,6 +77,15 @@ public ConfigDescription getConfigDescription() {
 
                     String typeUID = String.format("%s:%s", this.thingTypeUID.getBindingId(), typeId);
 
+                    int systemPrefixIdx = typeId.indexOf(SystemChannelTypeProvider.NAMESPACE_PREFIX);
+                    if (systemPrefixIdx != -1) {
+                        typeUID = String
+                                .format("%s:%s",
```

Comment:
Maybe you could reuse the method to construct a system channel type I mentioned in a previous comment here.